### PR TITLE
Feature/ddbteam 826 symfony upgrade

### PR DESCRIPTION
**Pull request by turegjorup**
#### Link to ticket

Please add a link to the ticket being addressed by this change.

#### Description

Please include a short description of the suggested change and the reasoning behind the approach you have chosen.

#### Screenshot of the result

If your change affects the user interface you should include a screenshot of the result with the pull request.

#### Checklist

- [x] My code passes our test (all our tests).
- [x] My code passes our static analysis suite.
- [x] My code passes our continuous integration process.

If your code does not pass all the requirements on the checklist you have to add a comment explaining why this change 
should be exempt from the list.

#### Additional comments or questions

If you have any further comments or questions for the reviewer please add them here.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -37,6 +37,7 @@ services:
       - mariadb
     volumes:
       - nfsApp:/app:delegated
+      - nfsApp2:/agency-auth-bundle:delegated
```
Comment (cableman) [inline docker-compose.yml]: That hack toke me 30 min to figure out: Cannot start service phpfpm: error while mounting volume '/var/lib/docker/volumes/cover-service-upload_nfsApp2/_data': failed to mount local volume

File: composer.json [hunk 2]
```diff
@@ -114,8 +131,15 @@
...
-            "require": "5.2.*",
+            "require": "6.1.*",
             "docker": false
         }
-    }
+    },
+    "repositories": [
+        {
+            "type": "path",
+            "url": "../agency-auth-bundle"
```
Comment (cableman) [inline composer.json]: Remember to remove before finale PR. :-)

File: docker-compose.yml (hunk 1 above)
Comment (turegjorup) [reply docker-compose.yml]: It's still a draft. Not ready for review ;-)

File: .env [hunk 3]
```diff
@@ -31,11 +32,10 @@ CORS_ALLOW_ORIGIN='^https?://(localhost|127\.0\.0\.1)(:[0-9]+)?$'
 # IMPORTANT: You MUST configure your server version, either here or in config/packages/doctrine.yaml
 #
 # DATABASE_URL="sqlite:///%kernel.project_dir%/var/data.db"
-# DATABASE_URL="mysql://db_user:db_password@127.0.0.1:3306/db_name?serverVersion=5.7"
+# DATABASE_URL="mysql://db_user:db_password@127.0.0.1:3306/db_name?serverVersion=5.7&charset=utf8mb4"
 # DATABASE_URL="postgresql://symfony:ChangeMe@127.0.0.1:5432/app?serverVersion=13&charset=utf8"
 DATABASE_URL=mysql://db:db@mariadb:3306/db
 DATABASE_SSL_CA='certs/combined.cert.pem'
```
Comment (cableman) [inline .env]: Out-side: but we should remove this line as the certs is not used anymore and have been deleted

File: .gitignore [hunk 4]
```diff
@@ -6,7 +6,6 @@
 /public/bundles/
 /var/
 /vendor/
-/public/cover
```
Comment (cableman) [inline .gitignore:9]: This should be re-add to ensure upload covers from local is not commit into github

File: .gitignore [hunk 5]
```diff
@@ -18,15 +17,15 @@
...
-
 ###> phpunit/phpunit ###
 /phpunit.xml
 .phpunit.result.cache
 ###< phpunit/phpunit ###
+
 node_modules
 .php_cs.cache
+
+###> symfony/phpunit-bridge ###
```
Comment (cableman) [inline .gitignore:23]: This is also located in line 21 above?

File: composer.json [hunk 6]
```diff
@@ -4,53 +4,61 @@
...
         "cloudinary/cloudinary_php": "^2.7",
-        "danskernesdigitalebibliotek/agency-auth-bundle": "^2.0",
+        "danskernesdigitalebibliotek/agency-auth-bundle": "^3.0",
         "doctrine/annotations": "^1.10",
         "doctrine/doctrine-bundle": "^2.5",
         "doctrine/doctrine-migrations-bundle": "^3.0",
         "doctrine/orm": "^2.11",
         "nelmio/cors-bundle": "^2.2",
         "phpdocumentor/reflection-docblock": "^5.3",
         "phpstan/phpdoc-parser": "^1.2",
```
Comment (cableman) [inline composer.json]: phpstan  package? Should it have been moved to required-dev?

Also phpdocumentor/reflection-docblock can be removed.

File: composer.json [hunk 7]
```diff
@@ -4,53 +4,61 @@
...
-        "symfony/console": "5.2.*",
-        "symfony/dotenv": "5.2.*",
-        "symfony/expression-language": "5.2.*",
-        "symfony/filesystem": "5.2.*",
-        "symfony/flex": "^1.3.1",
-        "symfony/framework-bundle": "5.2.*",
-        "symfony/http-client": "5.2.*",
-        "symfony/messenger": "5.2.*",
+        "predis/predis": "^2.0",
+        "psr/cache": "^3.0",
```
Comment (cableman) [inline composer.json]: Is symfony/cache not the only cache package needed?

File: composer.json [hunk 8]
```diff
@@ -4,53 +4,61 @@
...
+        "symfony/amqp-messenger": "6.1.*",
+        "symfony/asset": "6.1.*",
+        "symfony/cache": "6.1.*",
+        "symfony/console": "6.1.*",
+        "symfony/doctrine-messenger": "6.1.*",
+        "symfony/dotenv": "6.1.*",
+        "symfony/expression-language": "6.1.*",
+        "symfony/filesystem": "6.1.*",
+        "symfony/flex": "^2.2",
+        "symfony/form": "6.1.*",
```
Comment (cableman) [inline composer.json]: Do we need symfony/form?

File: composer.json [hunk 9]
```diff
@@ -4,53 +4,61 @@
...
-        "symfony/proxy-manager-bridge": "5.2.*",
-        "symfony/security-bundle": "5.2.*",
-        "symfony/serializer": "5.2.*",
-        "symfony/twig-bundle": "5.2.*",
-        "symfony/validator": "5.2.*",
-        "symfony/yaml": "5.2.*",
-        "vich/uploader-bundle": "^1.12"
+        "symfony/property-access": "6.1.*",
+        "symfony/property-info": "6.1.*",
+        "symfony/proxy-manager-bridge": "6.1.*",
```
Comment (cableman) [inline composer.json]: Do we need symfony/proxy-manager-bridge?

File: config/packages/api_platform.yaml [hunk 10]
```diff
@@ -32,9 +32,6 @@ api_platform:
     patch_formats:
         json: ['application/merge-patch+json']
 
-    swagger:
-        versions: [3]
```
Comment (cableman) [inline config/packages/api_platform.yaml:36]: Is it defaulting to version 3?

File: config/packages/monolog.yaml [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+        handlers:
+            main:
+                type: fingers_crossed
+                action_level: error
+                handler: nested
+                excluded_http_codes: [404, 405]
+                buffer_size: 50 # How many messages should be saved? Prevent memory leaks
+            nested:
+                type: stream
+                path: php://stderr
```
Comment (cableman) [inline config/packages/monolog.yaml]: Should be "php://stdout"....

File: config/packages/monolog.yaml [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+                level: debug
+                formatter: monolog.formatter.json
+            console:
+                type: console
+                process_psr_3_messages: false
+                channels: ["!event", "!doctrine"]
+            deprecation:
+                type: stream
+                channels: [deprecation]
+                path: php://stderr
```
Comment (cableman) [inline config/packages/monolog.yaml]: We should not log deprecation in production?

File: src/Command/CleanUpCommand.php [hunk 13]
```diff
@@ -10,38 +10,27 @@
...
-    private CoverService $coverStoreService;
-    private EntityManagerInterface $entityManager;
-
-    protected static $defaultName = 'app:image:cleanup';
-
     /**
      * CleanUpCommand constructor.
      */
-    public function __construct(CoverRepository $coverRepository, CoverService $coverStoreService, EntityManagerInterface $entityManager)
+    public function __construct(private readonly CoverRepository $coverRepository, private readonly CoverService $coverStoreService, private readonly EntityManagerInterface $entityManager)
```
Comment (cableman) [inline src/Command/CleanUpCommand.php]: Should be formatted correctly?

File: src/Command/CleanUpDatabaseCommand.php [hunk 14]
```diff
@@ -6,38 +6,27 @@
...
-    private CoverService $coverStoreService;
-    private EntityManagerInterface $em;
-
-    protected static $defaultName = 'app:database:cleanup';
-
     /**
      * CleanUpCommand constructor.
      */
-    public function __construct(CoverRepository $coverRepository, CoverService $coverStoreService, EntityManagerInterface $entityManager)
+    public function __construct(private readonly CoverRepository $coverRepository, private readonly CoverService $coverStoreService, private readonly EntityManagerInterface $em)
```
Comment (cableman) [inline src/Command/CleanUpDatabaseCommand.php]: Should be formatted correctly?

File: src/Command/CoverStoreSearchCommand.php [hunk 15]
```diff
@@ -8,29 +8,24 @@
...
-
-    private CoverStoreInterface $store;
-
     /**
      * CoverStoreSearchCommand constructor.
      *
      * @param CoverStoreInterface $store
      */
-    public function __construct(CoverStoreInterface $store)
+    public function __construct(private readonly CoverStoreInterface $store)
```
Comment (cableman) [inline src/Command/CoverStoreSearchCommand.php]: Should be formatted correctly?

File: src/Command/ExistsCommand.php [hunk 16]
```diff
@@ -6,38 +6,27 @@
...
-    private EntityManagerInterface $entityManager;
-    private MaterialRepository $materialRepository;
-
-    protected static $defaultName = 'app:cs:exists';
-
     /**
      * CleanUpCommand constructor.
      */
-    public function __construct(MaterialRepository $materialRepository, CoverService $coverStoreService, EntityManagerInterface $entityManager)
+    public function __construct(private readonly MaterialRepository $materialRepository, private readonly CoverService $coverStoreService, private readonly EntityManagerInterface $entityManager)
```
Comment (cableman) [inline src/Command/ExistsCommand.php]: Should be formatted correctly?

File: src/Command/GetRemoteUrlsCommand.php [hunk 17]
```diff
@@ -10,38 +10,27 @@
...
-    private CoverService $coverStoreService;
-    private EntityManagerInterface $entityManager;
-
-    protected static $defaultName = 'app:cover:get-remote-urls';
-
     /**
      * GetRemoteUrlsCommand constructor.
      */
-    public function __construct(CoverRepository $coverRepository, CoverService $coverStoreService, EntityManagerInterface $entityManager)
+    public function __construct(private readonly CoverRepository $coverRepository, private readonly CoverService $coverStoreService, private readonly EntityManagerInterface $entityManager)
```
Comment (cableman) [inline src/Command/GetRemoteUrlsCommand.php]: Should be formatted correctly?

File: src/Command/RequeueCommand.php [hunk 18]
```diff
@@ -22,40 +23,23 @@
...
-    private RouterInterface $router;
-    private EntityManagerInterface $em;
-
-    protected static $defaultName = 'app:image:requeue';
-
     /**
      * CleanUpCommand constructor.
      */
-    public function __construct(MaterialRepository $materialRepository, CoverService $coverStoreService, StorageInterface $storage, MessageBusInterface $bus, RouterInterface $router, EntityManagerInterface $entityManager)
+    public function __construct(private readonly MaterialRepository $materialRepository, private readonly CoverService $coverStoreService, private readonly StorageInterface $storage, private readonly MessageBusInterface $bus, private readonly RouterInterface $router, private readonly EntityManagerInterface $em)
```
Comment (cableman) [inline src/Command/RequeueCommand.php]: Should be formatted correctly?

File: src/DataFixtures/Faker/Provider/CoverProvider.php [hunk 19]
```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace App\DataFixtures\Faker\Provider;
+
+use Faker\Provider\Base;
+
+class CoverProvider extends Base
```
Comment (cableman) [inline src/DataFixtures/Faker/Provider/CoverProvider.php:7]: I'am missing some function documentation here as they a not inherited from Base

File: src/Doctrine/CurrentUserExtension.php [hunk 20]
```diff
@@ -20,17 +20,14 @@
...
-    private Security $security;
-
     /**
      * CurrentUserExtension constructor.
      *
      * @param Security $security
      *   The security service with the current user
      */
-    public function __construct(Security $security)
+    public function __construct(private readonly Security $security)
```
Comment (cableman) [inline src/Doctrine/CurrentUserExtension.php]: Should be formatted correctly?

File: src/EventSubscriber/MaterialPostWriteSubscriber.php [hunk 21]
```diff
@@ -22,25 +22,20 @@
...
-    private StorageInterface $storage;
-
     /**
      * MaterialPostWriteSubscriber constructor.
      *
      * @param MessageBusInterface $bus
      * @param StorageInterface $storage
      */
-    public function __construct(MessageBusInterface $bus, StorageInterface $storage)
+    public function __construct(private readonly MessageBusInterface $bus, private readonly StorageInterface $storage)
```
Comment (cableman) [inline src/EventSubscriber/MaterialPostWriteSubscriber.php]: Should be formatted correctly?

File: src/EventSubscriber/MaterialPreWriteSubscriber.php [hunk 22]
```diff
@@ -23,18 +23,16 @@
...
+    private readonly UserInterface $user;
 
     /**
      * MaterialPreWriteSubscriber constructor.
      *
      * @param MessageBusInterface $bus
      * @param Security $security
      */
-    public function __construct(MessageBusInterface $bus, Security $security)
+    public function __construct(private readonly MessageBusInterface $bus, Security $security)
```
Comment (cableman) [inline src/EventSubscriber/MaterialPreWriteSubscriber.php]: Should be formatted correctly?

File: src/EventSubscriber/ResolveCoverContentUrlSubscriber.php [hunk 23]
```diff
@@ -25,13 +25,8 @@
  */
 final class ResolveCoverContentUrlSubscriber implements EventSubscriberInterface
 {
-    private StorageInterface $storage;
-    private CoverService $coverStoreService;
-
-    public function __construct(StorageInterface $storage, CoverService $coverStoreService)
+    public function __construct(private readonly StorageInterface $storage, private readonly CoverService $coverStoreService)
```
Comment (cableman) [inline src/EventSubscriber/ResolveCoverContentUrlSubscriber.php]: Should be formatted correctly?

File: src/OpenApi/OpenApiDecorator.php [hunk 24]
```diff
@@ -13,22 +13,19 @@
...
 {
-    private NormalizerInterface $decorated;
-
     /**
      * OpenApiDecorator constructor.
      *
      * @param NormalizerInterface $decorated
      */
-    public function __construct(NormalizerInterface $decorated)
+    public function __construct(private readonly NormalizerInterface $decorated)
```
Comment (cableman) [inline src/OpenApi/OpenApiDecorator.php]: Should be formatted correctly?

File: src/OpenApi/OpenApiDecorator.php [hunk 25]
```diff
@@ -13,22 +13,19 @@
...
     {
-        $this->decorated = $decorated;
     }
 
     /**
      * {@inheritdoc}
      */
-    public function normalize($object, $format = null, array $context = [])
+    public function normalize($object, $format = null, array $context = []): array|string|int|float|bool|\ArrayObject|null
```
Comment (cableman) [inline src/OpenApi/OpenApiDecorator.php:29]: Could the return type be simplified: `mixed`

File: src/Repository/CoverRepository.php [hunk 26]
```diff
@@ -60,4 +60,32 @@ public function getNoRemoteUrl(int $limit = 0): Query
 
         return $queryBuilder->getQuery();
     }
+
+    /**
+     * Get query for all covers that are nor related to a Material.
```
Comment (cableman) [inline src/Repository/CoverRepository.php]: nor => not

File: src/Service/CoverService.php [hunk 27]
```diff
@@ -18,24 +18,15 @@
...
-
     /**
      * CoverStoreService constructor.
      *
      * @param StorageInterface $storage
      * @param Filesystem $filesystem
      * @param CoverStoreInterface $coverStore
      */
-    public function __construct(StorageInterface $storage, Filesystem $filesystem, CoverStoreInterface $coverStore, EntityManagerInterface $entityManager)
+    public function __construct(private readonly StorageInterface $storage, private readonly Filesystem $filesystem, private readonly CoverStoreInterface $coverStore, private readonly EntityManagerInterface $em)
```
Comment (cableman) [inline src/Service/CoverService.php]: Should be formatted correctly?

File: src/Service/CoverStore/CloudinaryCoverStoreService.php [hunk 28]
```diff
@@ -36,7 +35,7 @@ class CloudinaryCoverStoreService implements CoverStoreInterface
      *
      * @throws CoverStoreCredentialException
      */
-    public function __construct(string $bindCloudinaryCloudName, string $bindCloudinaryApiKey, string $bindCloudinaryApiSecret, string $bindCloudinaryFolder, int $bindCloudinarySearchTTL, AdapterInterface $cache)
+    public function __construct(string $bindCloudinaryCloudName, string $bindCloudinaryApiKey, string $bindCloudinaryApiSecret, string $bindCloudinaryFolder, int $bindCloudinarySearchTTL, private readonly CacheItemPoolInterface $cache)
```
Comment (cableman) [inline src/Service/CoverStore/CloudinaryCoverStoreService.php]: Should be formatted correctly?

File: src/Utils/Types/VendorState.php [hunk 29]
```diff
@@ -4,8 +4,8 @@
...
 class VendorState
 {
-    const INSERT = 'insert';
-    const UPDATE = 'update';
-    const DELETE = 'delete';
-    const DELETE_AND_UPDATE = 'deleteAndUpdate';
+    final public const INSERT = 'insert';
+    final public const UPDATE = 'update';
+    final public const DELETE = 'delete';
+    final public const DELETE_AND_UPDATE = 'deleteAndUpdate';
```
Comment (cableman) [inline src/Utils/Types/VendorState.php:10]: We should look into converting thes to enums (later both here and in imports :-))

**Review by cableman — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 4 above)
Comment (turegjorup) [reply .gitignore:9]: Done. 

(But it should never have been in the `###> symfony/framework-bundle ###` block)

File: .gitignore (hunk 5 above)
Comment (turegjorup) [reply .gitignore:23]: `###> vendor/package ###` blocks are added  by flex. Will leave as is even if they are duplicated

File: composer.json (hunk 7 above)
Comment (turegjorup) [reply composer.json]: `psr/cache` is only interfaces. Not implementation. Have remove direct dependency. Still required by other packages.

File: composer.json (hunk 9 above)
Comment (turegjorup) [reply composer.json]: No

File: composer.json (hunk 8 above)
Comment (turegjorup) [reply composer.json]: No

File: config/packages/api_platform.yaml (hunk 10 above)
Comment (turegjorup) [reply config/packages/api_platform.yaml:36]: "By default, OpenAPI v3 is used" - https://api-platform.com/docs/core/openapi/

**Review by cableman — APPROVED** [review]

**Pull request merged**